### PR TITLE
feat: [DHIS2-17655] Two event workspace

### PR DESCRIPTION
**Pull request by eirikhaugstulen**
Tech-summary:
- Adds a new widget available in the Enrollment Edit Event page
- The widget is self contained and checks if there are any linked events to the current event
- If there are any events, this gets formatted and displayed as a box on top of the current event workspace

image

### CODE REVIEW

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetTwoEventWorkspace.component.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+const WidgetTwoEventWorkspacePlain = ({ linkedEvent, dataValues, formFoundation, classes }: Props) => {
+    const dataEntryValues = useMemo(() => getDataEntryDetails(
+        linkedEvent,
+        formFoundation,
+    ), [linkedEvent, formFoundation]);
+
+    const listValues = useMemo(() => {
+        const elements = formFoundation.getElements();
+        const convertedValues = formFoundation.convertAndGroupBySection(dataValues, convertFn);
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/WidgetTwoEventWorkspace.component.js:39]: The dataElements of type `IMAGE` and `FILE` are not displayed. Can you check why the `PreviewImage` component and the file name URL are missing? Thanks!
Screenshot 2024-07-30 at 16 18 00

**Review by simonadomnisoru — COMMENTED** [review]
Hi @eirikhaugstulen,
It looks good overall 👏 . I added a comment with the dataElements types that are not properly displayed. Besides that, is it possible to add some cypress tests for this feature?
Thanks!

**Comment by eirikhaugstulen** [comment]
Thanks for the review @simonadomnisoru! I was missing the clientSubValues, which meant the values was not displaying correctly. Nice catch!

File: src/core_modules/capture-core/components/WidgetTwoEventWorkspace/utils/getSubValueForDataValue.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+const getFileResourceSubvalue = async ({ dataElement, querySingleResource, eventId, absoluteApiPath }: SubValueFunctionProps) => {
+    const { value } = dataElement;
+    if (!value) return null;
+
+    const { id, displayName: name } = await querySingleResource({ resource: `fileResources/${value}` });
+    return {
+        id,
+        name,
+        url: `${absoluteApiPath}/events/files?dataElementUid=${dataElement.id}&eventUid=${eventId}`,
```
Comment (simonadomnisoru) [inline src/core_modules/capture-core/components/WidgetTwoEventWorkspace/utils/getSubValueForDataValue.js:21]: A reminder that this URL is not working in `2.41`. The fix can be handled along with the rest of the places in [DHIS2-17537](https://dhis2.atlassian.net/browse/DHIS2-17537). Thanks!

**Review by simonadomnisoru — APPROVED** [review]

**Review by geethaalwan — APPROVED** [review]
Tested successfully on 2.41,2.41.1,2.40.5,2.39.7 versions

**Pull request merged**